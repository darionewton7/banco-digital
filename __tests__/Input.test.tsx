import React from 'react';
import renderer from 'react-test-renderer';
import CustomInput from '../src/components/Input';

// Mock onChangeText function
const mockOnChangeText = jest.fn();

describe('<CustomInput />', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <CustomInput 
        placeholder="Test Input"
        value=""
        onChangeText={mockOnChangeText}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with a label', () => {
    const tree = renderer.create(
      <CustomInput 
        label="Test Label"
        placeholder="Input with label"
        value=""
        onChangeText={mockOnChangeText}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with an error message', () => {
    const tree = renderer.create(
      <CustomInput 
        label="Input with Error"
        placeholder="Enter text"
        value="invalid input"
        onChangeText={mockOnChangeText}
        error="This field is required"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
    // Check if error style and error text are rendered
  });

  it('renders correctly with a predefined value', () => {
    const tree = renderer.create(
      <CustomInput 
        placeholder="Should have value"
        value="Predefined Value"
        onChangeText={mockOnChangeText}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // Testing onChangeText behavior requires a different testing library like @testing-library/react-native
});

