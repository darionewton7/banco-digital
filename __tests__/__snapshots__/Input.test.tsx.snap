// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomInput /> renders correctly with a label 1`] = `
<View
  style={
    {
      "marginBottom": 8,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#666666",
        "fontSize": 16,
        "marginBottom": 8,
      }
    }
  >
    Test Label
  </Text>
  <TextInput
    onChangeText={[MockFunction]}
    placeholder="Input with label"
    placeholderTextColor="#666666"
    style={
      [
        {
          "backgroundColor": "#ffffff",
          "borderColor": "#cccccc",
          "borderRadius": 8,
          "borderWidth": 1,
          "color": "#333333",
          "fontSize": 16,
          "height": 50,
          "marginBottom": 16,
          "paddingHorizontal": 16,
        },
        null,
        undefined,
      ]
    }
    value=""
  />
</View>
`;

exports[`<CustomInput /> renders correctly with a predefined value 1`] = `
<View
  style={
    {
      "marginBottom": 8,
      "width": "100%",
    }
  }
>
  <TextInput
    onChangeText={[MockFunction]}
    placeholder="Should have value"
    placeholderTextColor="#666666"
    style={
      [
        {
          "backgroundColor": "#ffffff",
          "borderColor": "#cccccc",
          "borderRadius": 8,
          "borderWidth": 1,
          "color": "#333333",
          "fontSize": 16,
          "height": 50,
          "marginBottom": 16,
          "paddingHorizontal": 16,
        },
        null,
        undefined,
      ]
    }
    value="Predefined Value"
  />
</View>
`;

exports[`<CustomInput /> renders correctly with an error message 1`] = `
<View
  style={
    {
      "marginBottom": 8,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#666666",
        "fontSize": 16,
        "marginBottom": 8,
      }
    }
  >
    Input with Error
  </Text>
  <TextInput
    onChangeText={[MockFunction]}
    placeholder="Enter text"
    placeholderTextColor="#666666"
    style={
      [
        {
          "backgroundColor": "#ffffff",
          "borderColor": "#cccccc",
          "borderRadius": 8,
          "borderWidth": 1,
          "color": "#333333",
          "fontSize": 16,
          "height": 50,
          "marginBottom": 16,
          "paddingHorizontal": 16,
        },
        {
          "borderColor": "#dc3545",
        },
        undefined,
      ]
    }
    value="invalid input"
  />
  <Text
    style={
      {
        "color": "#dc3545",
        "fontSize": 14,
        "marginBottom": 8,
        "marginTop": 4,
      }
    }
  >
    This field is required
  </Text>
</View>
`;

exports[`<CustomInput /> renders correctly with default props 1`] = `
<View
  style={
    {
      "marginBottom": 8,
      "width": "100%",
    }
  }
>
  <TextInput
    onChangeText={[MockFunction]}
    placeholder="Test Input"
    placeholderTextColor="#666666"
    style={
      [
        {
          "backgroundColor": "#ffffff",
          "borderColor": "#cccccc",
          "borderRadius": 8,
          "borderWidth": 1,
          "color": "#333333",
          "fontSize": 16,
          "height": 50,
          "marginBottom": 16,
          "paddingHorizontal": 16,
        },
        null,
        undefined,
      ]
    }
    value=""
  />
</View>
`;
